runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -275,8 +275,29 @@ ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda,
 	BOOLEAN newCPEntry = TRUE;
 	BuildResult result = OK;
 	char buf[ROM_ADDRESS_LENGTH + 1] = {0};
+	U_8 *hostPackageName = context->hostPackageName();
+	UDATA hostPackageLength = context->hostPackageLength();
 	PORT_ACCESS_FROM_PORT(_portLibrary);
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	if (context->isClassHidden() && !context->isHiddenClassOptStrongSet()) {
```

Comment:
Why the check of `!context->isHiddenClassOptStrongSet()` is here ?

Reply:
Because `InjectedInvoker` is a hidden !strong class as per the code which creates it: [MethodHandleImpl.java#L1161-L1168](https://github.com/ibmruntimes/openj9-openjdk-jdk16/blob/4da0cd5b3be77ac8f731bb5572bd343ebc104c65/src/java.base/share/classes/java/lang/invoke/MethodHandleImpl.java#L1161-L1168). By applying !strong, the number of `memcmp`s are reduced, which should minimize the impact on perf.

Reply:
Please add a comment in the code explaining that. Code here needs to be updated if things change on the JCL side. 

Is `CLASSOPTION_FLAG_NESTMATE` set ? Adding check of `isHiddenClassOptNestmateSet()` may help you filter out more cases.

Reply:
Verified that the `NESTMATE` attribute is not set for `InjectedInvoker.` Added `isHiddenClassOptNestmateSet()` to the filter. Added a comment to explain the above details.